src/data/list/basic.lean

```diff
@@ -2229,6 +2229,22 @@ begin
     apply hl _ (list.mem_cons_self _ _) }
 end
 
+lemma foldr_range_subset_of_range_subset {f : α → γ → γ} {g : β → γ → γ}
+  (hfg : set.range f ⊆ set.range g) (a) : set.range (foldr f a) ⊆ set.range (foldr g a) :=
```

Comment:
Is it the case that `set.range (foldr f a) = (submonoid.closure (range f) : submonoid (function.End γ))`?

Reply:
I can't get this statement to typecheck in Lean. But yes, `set.range (foldr f a)` is the union of `g a` for every `g` in the monoid generated by the indexed family `f`.

Should I prove this?

Reply:
It probably makes the dependencies difficult, so fine to do another time (if ever). It might be useful for your actual use-case though.